Certificate.py

```diff
@@ -0,0 +1,80 @@
+import json
+import ntpath
+from typing import Optional, Dict, List
+
+from Enum.JsonStructureKeys import JsonStructureKey
+from JsonItems.IJsonItem import IJsonItem
+
+
+# Class representation of certificate
+# it contains content, filename, content per page
+# content per page is guessed based on footers
+class Certificate:
+    def __init__(self, filename: str, content: str):
+        self._filename = ntpath.basename(filename)
+        # backup file name
+        if not self._filename:
+            self._filename = filename
+        self._content = content
+        self._pages = self._get_per_pages()
+        self._json = {}
+
+    def _get_per_pages(self) -> Dict[int, str]:
+        pages = {}
+        page_number = 1
+        splited_pages = self._content.split('\f')
+        for page_content in splited_pages:
+            if page_number < len(splited_pages):
+                pages[page_number] = page_content
+                page_number += 1
+        return pages
+
+    def get_pages_count(self) -> int:
+        return len(self._pages.keys())
+
+    def get_page_numbers(self) -> List[int]:
+        return list(self._pages.keys())
+
+    def get_page_content(self, page: int) -> str:
+        if page not in self._pages:
+            raise Exception("not valid page " + str(page) + " for file " + self._filename)
```

Comment:
Using Exception is too general, trying to catch this exception in the future would lead to catching basically any other exception.
It would be best if you threw a custom exception instead.
This issue happens multiple times in the code.